### PR TITLE
Campaign footnotes

### LINKED ISSUE #855: Campaign: Know It footnotes
Depends on #918 

The Solution Fact's footnote numbers need to start sequentially after the Problem Fact's footnotes.

e.g. Fact Problem has three sources <sup>123</sup>
Fact Solution has two sources <sup>45</sup>

Sources:
1- whatevs
2 - another source
3 - yea ok
4 - sure thing
5 - ok guy 

Create function to send an array of fact_fields and get the corresponding facts / footnotes, and sources arrays back. 
e.g. `get_fact_vars(array('field_fact_problem', 'field_fact_solution'));`

### PR DESCRIPTION
**Pull request by aaronschachter**
Fixes #855

Adds footnotes with corresponding sources into Campaign Actoin Page by way of a new  `dosomething_fact_get_mutiple_fact_field_vars($field_names)` function:

-- @param $field_names: an array of fact entityreference fields

-- @return: array of facts/footnotes and sources for all facts, indexed sequentially.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module [hunk 1]
```diff
@@ -98,7 +98,7 @@ function dosomething_helpers_add_entityref_links(&$node, $fields) {
 }
 
 /**
- * Implements hook_preprocess_html().
```
Comment (mmwtsn) [inline lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module:101]: :thumbsup:

File: lib/modules/dosomething/dosomething_fact/dosomething_fact.module [hunk 2]
```diff
@@ -34,6 +38,44 @@ function dosomething_fact_node_view($node, $view_mode, $langcode) {
...
+ *   -   @see dosomething_fact_get_fact_vars()
+ *   - sources: Sequential array of all sources found in all $field_names.
+ */
+function dosomething_fact_get_mutiple_fact_field_vars($node, $field_names) {
+  // Initalize return array.
+  $vars = array(
+    'facts' => array(),
+    'sources' => array(),
+  );
+  $wrapper = entity_metadata_wrapper('node', $node);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_fact/dosomething_fact.module:60]: does this black magic work if it's just an nid? :fireworks: :sake: :scream_cat:

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_fact/dosomething_fact.module:60]: It is definitely black magic and it does work with just a nid!  I can't find it documented but this works:

```
$node = entity_metadata_wrapper('node', 12);
print $node->title->value();
```

**Comment by mmwtsn** [comment]
Pulled it down and everything looks good. There is a duplicate source as Aaron noted in stand up.

facts

**Comment by mmwtsn** [comment]
:sailboat:

**Comment by angaither** [comment]
:+1:

**Pull request merged**